src/HoC.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {ensure} from './util';
+import {update, unmount} from './Mixin';
```

Comment:
For clarity below, I'd prefer `import Mixin from './Mixin'` and calling `Mixin.update(...)` and `Mixin.unmount(...)` below. If that requires refactoring Mixin, then it's not a big deal, don't worry about it.